File: backend/src/basico/002-ejercicios/002-suma_de_digitos/digits_sum.php
<?php

function digits_sum($number) {
    $response = "";
    $total = 0;
    while ($number != 0) {
        $rest = $number % 10;
        $response += (string)$rest;
        $total += $rest;
        $number = (int)($number/10);
        if ($number > 0) {
            $response += " + ";  // Las cadenas se pueden concatenar
                                 // mediante el operador de suma +
        }
    }
    $response += " = " + (string)$total;
    return $response;
}

$num1 = 2345;
$num2 = 326205;

echo digits_sum($num1), "\n";
echo digits_sum($num2), "\n";

?>

File: backend/src/basico/002-ejercicios/002-suma_de_digitos/digits_sum.go
package main

import "strconv"

var num1 int = 4511
var num2 int = 369301

/* Con la función ``strconv.Itoa()`` convertimos
    los números enteros en cadenas. */

func digits_sum(number int) (string) {
    var response string = ""
    var total int = 0
    for {
        if number == 0 { break }
        rest := number % 10
        response += strconv.Itoa(rest)
        total += rest
        number = int(number/10)
        if number > 0 {
            response += " + " // Las cadenas se pueden
        }                     // concatenar con el operador de suma
    }
    response += " = " + strconv.Itoa(total)
    return response
}

func main() {
    println(digits_sum(num1))
    println(digits_sum(num2))
}

/* Fuentes:
https://stackoverflow.com/questions/10105935/how-to-convert-an-int-value-to-string-in-go/10105983#10105983
*/
